clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -82,20 +100,28 @@ export class CopybookDownloadService implements vscode.Disposable {
                     await this.handleQueue(element, errors, progress);
 
                     if (this.queue.length === 0 && errors.size > 0) {
-                        this.createErrorMessageForCopybooks(errors);
+                        await CopybookDownloadService.createErrorMessageForCopybooks(errors);
                         errors.clear();
                     }
 
                     if (this.queue.length === 0) {
                         TelemetryService.registerEvent("Download copybooks from MF", ["copybook", "COBOL", "experiment-tag"], "total time to search copybooks on MF", new Map().set("time elapsed", TelemetryService.calculateTimeElapsed(startTime, Date.now())));
-                        startTime = 0;
+                        startTime = null;
                     }
                 });
         }
     }
 
-    private createErrorMessageForCopybooks(datasets: Set<string>) {
-        this.resolver.processDownloadError(PROCESS_DOWNLOAD_ERROR_MSG + Array.from(datasets));
+    private static async createErrorMessageForCopybooks(datasets: Set<string>) {
+        await CopybookDownloadService.processDownloadError(PROCESS_DOWNLOAD_ERROR_MSG + Array.from(datasets));
```

Comment:
what about to collapse one level and put there the content of ```processDownloadError()```?

Reply:
`processDownloadError()` is also used in `fetchCopybook`. So, I need to put a procedure content into two places. 
Unlike the `showQueueLockedDialog` function, which I use only once.